### PR TITLE
build: Remove unused dependencies

### PR DESCRIPTION
**Pull request by anonrig**
I've given @boshen 's cargo-shear a try: https://github.com/Boshen/cargo-shear

#skip-changelog

### CODE REVIEW

File: relay-dynamic-config/Cargo.toml [hunk 1]
```diff
@@ -30,9 +30,6 @@ relay-quotas = { workspace = true }
 relay-sampling = { workspace = true }
 serde = { workspace = true }
 serde_json = { workspace = true }
-smallvec = { workspace = true }
 
 [dev-dependencies]
-insta = { workspace = true }
```
Comment (colin-sentry) [inline relay-dynamic-config/Cargo.toml:36]: this is probably required

Comment (colin-sentry) [reply relay-dynamic-config/Cargo.toml:36]: for `cargo insta review`

Comment (anonrig) [reply relay-dynamic-config/Cargo.toml:36]: I couldn't find any usages inside `relay-dynamic-config`. Is this line still required to run `cargo insta review` even if the workspace has `insta` as a dependency?

**Review by Dav1dde — APPROVED** [review]

File: relay-dynamic-config/Cargo.toml (hunk 1 above)
Comment (jan-auer) [reply relay-dynamic-config/Cargo.toml:36]: `cargo insta` commands require the crate installed on the host, so it doesn't have to be in dependencies.

**Pull request merged**